### PR TITLE
Remove useless smarty caching

### LINKED ISSUE #19151: Remove cache on module ps_sharebuttons
Please remove cache from this module. This is useless:

https://github.com/PrestaShop/ps_sharebuttons/blob/master/ps_sharebuttons.php#L145

There is nothing to cache, and because of complexity of cache key we have thousands more files of cache worth nothing...

### PR DESCRIPTION
**Pull request by kpodemski**
There's literally nothing to cache here, redundant cache files all over the /cache folder

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | cache is useless in this module
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#19151.
| How to test?  | No behavior change. Module should work as before.
| Sponsor company  | impSolutions

### CODE REVIEW

**Comment by matks** [comment]
Hi ! If I understand correctly, the cache key is based on different product IDs however since the cached template does not differ for one or another product, it means we create cache files for Smarty for each cache ID but the cached content is always the same. Is that right ?

However this would be useful
- if one day we use product data to render different HTML
- if someone overrides into his theme this template and use product data

**Comment by kpodemski** [comment]
@matks 

> it means we create cache files for Smarty for each cache ID but the cached content is always the same. Is that right ?

no, content is different **but** there is nothing to cache, you don't perform any action which needs to be cached, and if someone has big catalog of the products, it creates files for all products and their combinations which is **useless** because **we have nothing to cache in this module**

every action performed in getWidgetVariables is eaither:
- cached by default (like getCover)
- or light, like getting an URL for a product...

**Comment by kpodemski** [comment]
@Progi1984 @matks bump

**Comment by matks** [comment]
> @matks
> 
> > it means we create cache files for Smarty for each cache ID but the cached content is always the same. Is that right ?
> 
> no, content is different

OK, now that I read the code again I see it.

> because we have nothing to cache in this module

I am sorry I dont understand this statement. Yes we have something to cache: Smarty templates. Removing a compilation step is always good for rendering. There is no faster code than the one that is not executed (at runtime) ? 😄 

> every action performed in getWidgetVariables is either:
> 
> * cached by default (like getCover)
> * or light, like getting an URL for a product...

When I check `Link::getProductLink()` I can see that there is some logic there. Not much, but still some logic and it includes loading objects from database, which is the most consuming step in php nowadays I believe (since php7 awesome optimization, the database connection is now the main bottleneck).


So indeed this part is not the most critical part fo optimize with cache ... but is it really an issue to keep cache enabled here ? All the Smarty cache files created are not loaded simultaneously.

I might miss something but it looks cache is is a unneeded bonus, but a bonus nevertheless 😄 and I actually like the idea of promoting the following best practice "whatever is done on FO must be cached by Smarty as much as possible" (and even better: put it in Varnish ! but that's another story)

I fail to see the advantage of removing the cache here but I might be lacking something 😉 please allow me to ping colleagues to help me understand
ping @Quetzacoalt91 @jolelievre @matthieu-rolland

**Comment by Matt75** [comment]
Well in fact there are an issue with this module : it generate too much cache files...
On shared hosting with IO limitation or inodes limitation, cache going too big to stay under limit.

Current cache key : ps_cashondelivery + PS_SSL_ENABLED + Shop identifier + Customer **Groups** identifier + Language identifier + Currency Identifier + Country identifier + Product identifier + Product Attribute identifier

So now you can calculate how many cache files this module will generate for a Shop with 1000 products, 250 product attributes, 3 Customer Groups, 5 Languages, 3 Currencies, All countries activated

Maybe it's too much ?
And there are some others modules who generate a lot of cache files...

**Comment by Quetzacoalt91** [comment]
I remember removing equivalent cache on other modules. We were caching too hard, which was making the content displayed always obsolete, thus worthless.

I even had some issues with content displayed in the wrong language, because it was common to all langs.

**Comment by matks** [comment]
Thanks @Matt75 and @Quetzacoalt91 for feedback 😉 then my conclusion is that @kpodemski PR is very valid

I will try now for every module to check whether the caching strategy is too aggressive, like this one. But the very best would be to have some "guidelines" about "relevant caching strategy" telling us what kind of cache level should be provided for which kind of template and data.

Let's go for QA 💪

**Review by matks — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @kpodemski & @sarahdib

**Comment by PrestaworksJonas** [comment]
Hi, is it possible for you guys to merge the dev branch with the master so this fix gets included in the latest PS versions? We have had a couple of customers having problem with this exact issue where too many cache files are being generated because of the faulty cache id.

@matks @PierreRambaud

**Comment by kpodemski** [comment]
ping @PrestaShop/prestashop-maintainers 

we could release it

**Comment by matks** [comment]
Hi @kpodemski well now that you are a maintainer, you dont need our approval to release a new version of a module 😄 

The only thing you need to do is
- open a PR to bump version number to 2.1.1, no need to validate by QA
- then open a PR to merge `dev` into `master, this one needs to be validated by QA (they will see it in the queue when we add the label "Waiting for QA" but we can also ping them on the Slack channel)

See https://devdocs.prestashop.com/1.7/project/release/release-a-native-ps-module/

Do you want to try following this process, as a first attempt 😊 ?

I just updated [milestone 2.1.1](https://github.com/PrestaShop/ps_sharebuttons/milestone/1).

**Pull request merged**
